### PR TITLE
Add `--cache-last-block` flag

### LINKED ISSUE #1598: add --cache-last-block flag description
The flag --cache-last-block created in this PR https://github.com/hyperledger/besu/pull/6009 is not yet documented.
This flags makes Besu cache n blocks data (block headers, block bodies, transactions' receipts and total difficulty).

The default number of blocks to keep in the cache is 0.

This PR will mainly improve the performances of some RPC calls :

Eth_getBlockByNumber
Eth_getBlockByHash
Eth_feeHistory
Eth_getTransactionReceipt
...
The biggest improvement is on Eth_feeHistory.

The new cache in DefaultBlockchain uses 1.35 GiB on Mainnet for 2048 last blocks :

Transactions receipts : 676 MiB
Blocks' bodies : 555 MiB
Blocks' headers : 2 MiB
Total difficulty : less than 1 MiB

### PR DESCRIPTION
**Pull request by joaniekube**
Fixes #1598

Add `--cache-last-block` flag description

### CODE REVIEW

**Review by ahamlat — APPROVED** [review]
LGTM

File: docs/public-networks/reference/cli/options.md [hunk 1]
```diff
@@ -408,6 +408,48 @@ When connecting to Mainnet or public testnets, the default is a predefined list
...
+```bash
+cache-last-blocks=2048
+```
+
+</TabItem>
+
+</Tabs>
+
+Configures the number of recent blocks to cache, significantly improving the performance of several RPC calls. 
+This caching significantly improves the performance of several RPC calls, particularly [`eth_getBlockByNumber`](../api/index.md#eth_getBlockByNumber), [`eth_getBlockByHash`](../api/index.md#eth_getBlockByHash), [`eth_getTransactionReceipt`](../api/index.md#getTransactionReceipt), and especially [`eth_feeHistory`](../api/index.md#eth_feeHistory).
```
Comment (alexandratran) [inline docs/public-networks/reference/cli/options.md:450]:
```suggestion
The number of recent blocks to cache. 
Using this option can improve the performance of several RPC calls: [`eth_getBlockByNumber`](../api/index.md#eth_getblockbynumber), [`eth_getBlockByHash`](../api/index.md#eth_getblockbyhash), [`eth_getTransactionReceipt`](../api/index.md#gettransactionreceipt), and especially [`eth_feeHistory`](../api/index.md#eth_feehistory). 
```

Edit for consistency with other option descriptions and to remove wording redundancy.

Also, please always check that your links work properly in the PR preview: https://besu-docs-git-1598-add-flag-hyperledger.vercel.app/development/public-networks/reference/cli/options#cache-last-blocks

Markdown requires heading anchor links to be all lowercase.

**Review by garyschulte — COMMENTED** [review]
LGTM, pending handling pending feedback 👍

File: docs/public-networks/reference/cli/options.md (hunk 1 above)
Comment (joaniefromtheblock) [reply docs/public-networks/reference/cli/options.md:450]: @alexandratran  Would it be alright to add the word "significantly" to this? The initial PR highlights a substantial improvement in performance, and I believe including this word would more accurately reflect the extent of the enhancement.

Comment (alexandratran) [reply docs/public-networks/reference/cli/options.md:450]: "Significantly" is a relative term that, IMO, doesn't add any more information to the sentence. I think either specify the "significant improvement" in quantitative terms or leave out the relative term. But not a big deal to use it either so it's up to you.

**Review by alexandratran — APPROVED** [review]
LGTM

**Pull request merged**